int16 theta1
int16 theta2
int16 theta3
float64 x
float64 z
float64 alpha
---
float64[] theta1s
float64[] theta2s
float64[] theta3s